aries/aries_vcx_core/src/wallet/agency_client_wallet.rs

```diff
@@ -1,166 +1,132 @@
-use std::{collections::HashMap, sync::Arc};
+use std::sync::Arc;
 
 use agency_client::{
     errors::error::{AgencyClientError, AgencyClientErrorKind, AgencyClientResult},
     wallet::base_agency_client_wallet::BaseAgencyClientWallet,
 };
 use async_trait::async_trait;
-#[cfg(feature = "vdrtools_wallet")]
-use vdrtools::WalletHandle;
+use public_key::{Key, KeyType};
 
-use super::structs_io::UnpackMessageOutput;
+use super::{
+    base_wallet::{BaseWallet, DidData, DidWallet, Record, RecordWallet, SearchFilter},
+    structs_io::UnpackMessageOutput,
+};
 use crate::{
     errors::error::{AriesVcxCoreError, AriesVcxCoreErrorKind, VcxCoreResult},
-    utils::async_fn_iterator::AsyncFnIterator,
-    wallet::base_wallet::BaseWallet,
+    wallet::entry_tag::EntryTags,
 };
 
 #[derive(Debug)]
 pub struct AgencyClientWallet {
     inner: Arc<dyn BaseAgencyClientWallet>,
 }
 
-/// Implementation of [BaseWallet] for [AgencyClientWallet] such that a [BaseAgencyClientWallet]
-/// can be converted to a [BaseWallet] for packing and unpacking messages, and vice versa.
+impl BaseWallet for AgencyClientWallet {}
+
 #[allow(unused_variables)]
 #[async_trait]
-impl BaseWallet for AgencyClientWallet {
-    async fn create_and_store_my_did(
-        &self,
-        seed: Option<&str>,
-        method_name: Option<&str>,
-    ) -> VcxCoreResult<(String, String)> {
-        Err(unimplemented_agency_client_wallet_method(
-            "create_and_store_my_did",
-        ))
-    }
-
-    async fn key_for_local_did(&self, did: &str) -> VcxCoreResult<String> {
-        Err(unimplemented_agency_client_wallet_method(
-            "get_verkey_from_wallet",
-        ))
-    }
-
-    async fn replace_did_keys_start(&self, target_did: &str) -> VcxCoreResult<String> {
-        Err(unimplemented_agency_client_wallet_method(
-            "replace_did_keys_start",
-        ))
+impl RecordWallet for AgencyClientWallet {
```

Comment:
Non actionable here

Just wondering how difficult it would be to remove this. I have feeling that pretty easy now, that `AgencyClientWallet` was originally conceived to avoid cyclical dependency between `agency_client` and `aries_vcx`, but that would not longer be the case given the wallet stuff was take out to `aries_vcx_core`.